### ISSUE TITLE
Bug: Lambda function not tracing the code

### ISSUE THREAD

**Issue by apaidipa** [aws-powertools/powertools-lambda-typescript#745]
## Bug description
Tracer not capturing the annotations and metadata inside lambda function, However it does capture the top level lambda traces. 

### Expected Behavior
Trace metadata, annotations inside my lambda function. 

### Current Behavior
Currently only top level lambda traces are captured.

### Environment

* **Powertools version used**: 0.7.2
* **Packaging format (Layers, npm)**: Yarn
* **AWS Lambda function runtime:**  NODEJS_14_X
* **Debugging logs:**

**Comment by dreamorosi** [comment]
Could you please provide a minimal example of your function code to reproduce the issue? I'm interested especially in seeing how you have instrumented the function.

**Comment by dreamorosi** [comment]
@apaidipa (also AWS employee) shared the code privately as it was close source and could not be shared publicly.

I was able to troubleshoot the issue and conclude that it was not in fact, a bug of Tracer but rather an issue with the implementation.

To benefit future readers, this particular implementation was using [manual instrumentation](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/tracer/#lambda-handler). When closing the `subsegment`, it had a conditional statement that checks whether the subsegment is already closed:
```ts
if (subsegment && !subsegment.isClosed) { // <- DO NOT COPY THIS, THIS WON'T WORK, KEEP READING
  subsegment.close();
}
```
the code above should have been instead:
```ts
if (subsegment && !subsegment.isClosed()) { // <- subsegment.isClosed() is a function that returns a boolean value
  subsegment.close();
}
```

Closing the issue as resolved.
